### PR TITLE
upload_s3: improve tags handling

### PR DESCRIPTION
**Pull request by cole-h**

### CODE REVIEW

**Comment by cole-h** [comment]
(WIP, cuz I still need to validate this does the right thing, and I didn't want to risk some actions not running on a draft PR)

**Comment by cole-h** [comment]
OK, validated -- LGTM I think. The tag script appropriately references the `install.det.sys` tag install URL, and the commit for the PR as well as the PR install URL are the same. Deleting the test tag and draft release for it.

**Comment by cole-h** [comment]
OK, validated for real this time. Good to go.

**Review by Hoverbear — APPROVED** [review]

File: upload_s3.sh [hunk 1]
```diff
@@ -2,37 +2,70 @@ set -eu
...
+
+  for file in $(find "$dest" -type f); do
+    artifact_path="$dest"/"$(basename "$artifact")"
+    md5="$(md5sum "$artifact" | cut -d' ' -f1)"
+    obj="$(aws s3api head-object --bucket "$AWS_BUCKET" --key "$artifact_path" || echo '{}')"
+    obj_md5="$(jq -r .ETag <<<"$obj" | jq -r)" # head-object call returns ETag quoted, so `jq -r` again to unquote it
+
+    if [[ "$md5" == "$obj_md5" ]]; then
+      echo "Artifact $artifact was already uploaded; exiting"
+      # If we already uploaded to a tag, that's probably bad
```
[the remark this answers is not in the document]
Comment (cole-h) [reply upload_s3.sh:57]: You could just force-push the tag instead of deleting, but yeah.

**Pull request merged**